### PR TITLE
New rule: avoid_classes_with_only_static_members

### PR DESCRIPTION
**Pull request by JPaulsen**
Fixes https://github.com/dart-lang/linter/issues/539

### CODE REVIEW

File: lib/src/rules/avoid_defining_a_class_that_contains_only_static_members.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    return classMember.isStatic;
+  }
+  return false;
+}
+
+class AvoidDefiningAClassThatContainsOnlyStaticMembers extends LintRule {
+  _Visitor _visitor;
+  AvoidDefiningAClassThatContainsOnlyStaticMembers()
+      : super(
+            name: 'avoid_defining_a_class_that_contains_only_static_members',
```
Comment (bwilkerson) [inline lib/src/rules/avoid_defining_a_class_that_contains_only_static_members.dart]: Perhaps we can find a shorter name, such as 'avoid_classes_with_only_static_members' (although something even shorter would be nice).

Comment (JPaulsen) [inline lib/src/rules/avoid_defining_a_class_that_contains_only_static_members.dart]: Fixed

**Comment by JPaulsen** [comment]
@bwilkerson yes, I have run all my pull requests in the SDK since my last bug in do_not_check_for_null_in_equality_operators

**Comment by bwilkerson** [comment]
How many lints were generated?

**Comment by JPaulsen** [comment]
@bwilkerson 96 in the SDK, they are all classes with at least one non-const static field, and only static methods

File: test/rules/avoid_defining_a_class_that_contains_only_static_members.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// test w/ `pub run test -N avoid_defining_a_class_that_contains_only_static_members`
+
+class Bad { // LINT
+  static int a;
+
+  static foo() {}
+}
+
+class Bad2 extends Good1 { // LINT
```
Comment (JPaulsen) [inline test/rules/avoid_defining_a_class_that_contains_only_static_members.dart]: Yes, but every class extends Object and it has nonstatic members (like toString), so I thinks the spirit of the rule, is not to create classes that locally only have static members.

**Comment by Nathan-Heller** [comment]
> @bwilkerson 96 in the SDK, they are all classes with at least one non-const static field, and only static methods

Is there a specific reason that the rule isn't violated when a class contains only static const fields? What was the rationale there?

Thanks for your time 😄

**Pull request merged**